File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String S, int k) {
       int[] frq = new int[26];
        int mxCnt = 0;
        int mxLen = 0;
        int j = 0;
        for(int i = 0;i<S.length();i++){
            frq[S.charAt(i)-'A']++;
            mxCnt = Math.max(mxCnt,frq[S.charAt(i)-'A']);
            if(i-j+1-mxCnt>k){
                frq[S.charAt(j)-'A']--;
                j++;
            }
            mxLen = Math.max(mxLen,i-j+1);
        }
        return mxLen;
    }
}

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int i =0;
        int j = 0;
        int cnt = 0;
        int ans = 0;
        map<char,int> mp;
        while(j<s.length()){
            mp[s[j]]++;
            cnt = max(cnt,mp[s[j]]);
            if(j-i+1 - cnt > k){
                mp[s[i]]--;
                i++;
            }
            ans = max(ans,j-i+1);
            j++;
        }
        return ans;
    }
};